Create dropin-starter-template.md

supernova-at opened the pull request:
Creates the initial draft for using the dropin starter template.

- [ ] Need to verify the steps in this tutorial work for others
- [ ] Need to find a home on the docs site for this to live

supernova-at committed:
Create dropin-starter-template.md

Creates the initial draft for using the dropin starter template.

jcalcaben commented:
> Should we instruct them to add a Container first and put the shared Button component there?

Yes, they should create a container first

```
npx elsie generate container --pathname <MyContainer>
```

Then a UI component

```
npx elsie generate container --pathname <MyContainer>
```

> how do I get my newly created Container to show up in the "Frontend Containers" area?

Update their sandbox environment to render their container that has the UI component (`examples/html-host/index.html`)

jcalcaben commented:
@jeff-matthews or @bdenham 

Is this content publicly available anywhere? https://elsie.dev/docs/

I see that it's still behind a permission wall thats based on GitHub users. If not, how would external developers know how to work with the SDK components?

jcalcaben commented on _drafts/dropin-starter-template.md:
```diff
@@ -0,0 +1,52 @@
+1. **Navigate to the Template Repository**: Go to https://github.com/adobe-commerce/dropin-template.
+2. **Create a New Repository**: Click on the "Use this template" button to create a new repository based on the template. This will generate a new repository with the same directory structure and files as the template.
+3. **Clone Your New Repository**: You can now clone the newly created repository to your local machine using `git clone`.
+4. **Install, Configure, and Launch**: Follow the instructions in your new repo's `README` file to install, configure, and launch your development environment.
```
**suggestion:** This should probably be more explicit with the actual commands from the README. Also, it's missing the important step of generating the elsie config file.

supernova-at replied:
I kind of purposefully didn't duplicate the content in case it changes. I can be more explicit about the steps though.

jeff-matthews commented:
> @jeff-matthews or @bdenham
> 
> Is this content publicly available anywhere? https://elsie.dev/docs/
> 
> I see that it's still behind a permission wall thats based on GitHub users. If not, how would external developers know how to work with the SDK components?

@jcalcaben, it's not public yet. we plan to integrate the SDK storybook output from the elsie site with the [current docs site](https://experienceleague.adobe.com/developer/commerce/storefront/) to create a unified experience. last i heard, we need to do this for the April 8th marketing release window. 

i assume that also means we should figure out how the content in this PR fits with that. @bdenham is going to look into it when he wraps up the summit lab workbook.

jcalcaben commented on _drafts/dropin-starter-template.md:
```diff
@@ -0,0 +1,52 @@
+
+To add a shared component from the Adobe Commerce Storefront SDK to your new project, follow these steps:
+
+1. **Install the `@dropins/tools` Package**: Run the following command to install the Storefront SDK package:
```
**suggestion:** This is the correct package for dropin development. The other one is for storefront

```suggestion
1. **Install the `@adobe-commerce/elsie ` Package**: Run the following command to install the Storefront SDK package:
```

supernova-at committed:
Update _drafts/dropin-starter-template.md

Co-authored-by: James Calcaben <jcalcaben@users.noreply.github.com>

supernova-at committed:
Update dropin-starter-template.md

Updates instructions to include generating a component, a container, and adding the container to the sandbox env.

supernova-at committed:
Merge branch 'develop' into patch-1

jcalcaben reviewed (approved):
Overall this is good enough as a starting point. We'll probably need additional in depth guides on dropin development best practices.

I think we need to clean up the sandbox index.html file so that there is less to remove or comment out to get things working, but that can be its own PR.

bdenham commented on _drafts/dropin-starter-template.md:
```diff
@@ -0,0 +1,63 @@
+
+## What are GitHub Templates?
+
+The Drop-in Template repository is a GitHub Template repository. GitHub Templates allow you to create new repositories with the same directory structure, branches, and files as an existing repository. This feature is particularly useful for setting up new projects quickly and consistently.
```
```suggestion
Drop-in templates are GitHub Templates that allow you to quickly create drop-in components with the same structure, branches, files, and best practices built-in. The `dropin-template` repository provides the starting point for creating new drop-ins quickly and consistently.
```

supernova-at replied:
Is "drop-in components" correct here? 

We have drop-in applications ("drop-ins") and then those applications have UI components in them, which I would call "drop-in components". 

I'm trying to refer to the application, not a UI component. Do we call the applications "drop-in components" elsewhere?

bdenham commented on _drafts/dropin-starter-template.md:
```diff
@@ -0,0 +1,63 @@
+
+This guide helps developers and partner agencies quickly create their own drop-in applications by leveraging Adobe Commerce's Drop-in Template repository.
+
+## What are GitHub Templates?
```
```suggestion
## What are drop-in component templates?
```

bdenham commented on _drafts/dropin-starter-template.md:
```diff
@@ -0,0 +1,63 @@
+# Using the Adobe Commerce Drop-in Template Repository
+
+This guide helps developers and partner agencies quickly create their own drop-in applications by leveraging Adobe Commerce's Drop-in Template repository.
```
```suggestion
This topic describes how to use the `drop-template` repository to create drop-in components for Adobe Commerce Storefronts.
```

supernova-at replied:
Same question here regarding the term "drop-in components"

bdenham commented on _drafts/dropin-starter-template.md:
```diff
@@ -0,0 +1,63 @@
+# Using the Adobe Commerce Drop-in Template Repository
```
```suggestion
# Getting started with drop-in templates
```

bdenham commented on _drafts/dropin-starter-template.md:
```diff
@@ -0,0 +1,63 @@
+
+6. **Render the Container**: Update your sandbox environment to render your container in `examples/html-host/index.html`.
+
+## Conclusion
```
I would remove the Conclusion section. It doesn't provide any more information. And we'll provide a general support area with links somewhere else in the docs.

bdenham reviewed (approved):
Approved with acceptance of suggested changes. Nice job, Andy!

supernova-at committed:
Update _drafts/dropin-starter-template.md

Co-authored-by: Bruce Denham <brucedenham@mac.com>

supernova-at committed:
Update _drafts/dropin-starter-template.md

Co-authored-by: Bruce Denham <brucedenham@mac.com>

supernova-at committed:
Update dropin-starter-template.md

Removes "conclusion" section

devpatil7 commented:
@supernova-at Please add supported Node versions as Maintenance (20) and Active (22)

supernova-at committed:
Update dropin-starter-template.md

Incorporates change suggestions

supernova-at commented on _drafts/dropin-starter-template.md:
```diff
@@ -0,0 +1,61 @@
+
+## How to Use the Adobe Commerce Drop-in Template
+
+Note: Supported Node versions are: Maintenance (v20) and Active (v22).
```
Note: new content added here per https://github.com/commerce-docs/microsite-commerce-storefront/pull/212#issuecomment-2724952629.

supernova-at committed:
Apply suggestions from code review

Incorporates suggested fixes

Co-authored-by: Bruce Denham <brucedenham@mac.com>

bdenham committed:
Merge branch 'develop' into patch-1
